packages/language/src/parser/parser.ts

```diff
@@ -0,0 +1,6536 @@
+        }
+      }
+    }
+
+    state.consume(
+      element,
+      CstNodeKind.ProcedureCallArgs_CloseParen,
+      tokens.CloseParen,
+    );
+
+    return element;
+  },
+);
+
+const labelReference = rule(
+  sequence(tokens.ID),
+  (state: ParserState): ast.LabelReference => {
+    const element = ast.createLabelReference();
+
+    const idToken = state.consume(
+      element,
+      CstNodeKind.LabelReference_LabelRef,
+      tokens.ID,
+    );
+    if (idToken) {
+      element.label = ast.createReference(
+        element,
+        idToken,
+        ast.ReferenceType.Variable,
+      );
+    }
+
+    return element;
+  },
+);
+
+const unaryExpression = rule(
+  sequence(tokens.UnaryOperator),
+  (state: ParserState): ast.UnaryExpression => {
+    const element = ast.createUnaryExpression();
+
+    const operatorToken = state.consume(
+      element,
+      CstNodeKind.UnaryExpression_Operator,
+      tokens.UnaryOperator,
+    );
+    if (operatorToken) {
+      element.op = tokens.UnaryOperator.mapToEnumLiteral(
+        operatorToken.tokenTypeIdx,
+      );
+    }
+
+    element.expr = expression.rule(state);
+
+    return element;
+  },
+);
+
+const literal = rule(
+  () => literalValue.first(),
+  (state: ParserState): ast.Literal => {
+    const element = ast.createLiteral();
+    element.value = literalValue.rule(state);
+    return element;
+  },
+);
+
+const literalValue = orRule<ast.LiteralValue>(
+  () => stringLiteral,
+  () => numberLiteral,
+);
+
+const stringLiteral = rule(
+  sequence(tokens.STRING_TERM),
+  (state: ParserState): ast.StringLiteral => {
+    const element = ast.createStringLiteral();
+
+    const stringToken = state.consume(
+      element,
+      CstNodeKind.StringLiteral_ValueString,
+      tokens.STRING_TERM,
+    );
+    if (stringToken) {
+      element.value = stringToken.image;
+    }
+
+    return element;
+  },
+);
+
+const numberLiteral = rule(
+  sequence(tokens.NUMBER),
+  (state: ParserState): ast.NumberLiteral => {
+    const element = ast.createNumberLiteral();
+
+    const numberToken = state.consume(
+      element,
+      CstNodeKind.NumberLiteral_ValueNumber,
+      tokens.NUMBER,
+    );
+    if (numberToken) {
+      element.value = numberToken.image;
+    }
+
+    return element;
+  },
+);
+
+export function performAssignmentLookahead(state: ParserState): boolean {
+  const expressionTokenTypes = [
+    tokens.ID,
+    tokens.BinaryOperator,
+    tokens.UnaryOperator,
+    tokens.AssignmentOperator,
+    tokens.STRING_TERM,
+    tokens.NUMBER,
+    tokens.Comma,
+    tokens.Dot,
+  ];
+
+  let previousToken: tokens.Token | undefined = undefined;
+  const lookahead: (la: number) => tokens.Token | undefined = (la) => {
+    previousToken = state.peek(la - 1);
+    return state.peek(la);
+  };
+  let i = 1;
+  let token = lookahead(i++);
+  // First token of an assigment needs to be an ID
+  if (!token || !tokenMatcher(token, tokens.ID)) {
+    return false;
+  }
```

Comment:
Suggestion: If the token is of token `ID` (but not a keyword!) we should be able to directly return true here:
```suggestion
  // First token of an assigment needs to be an ID
  if (!token || !tokenMatcher(token, tokens.ID)) {
    return false;
  } else if (token.tokenTypeIdx === tokens.ID.tokenTypeIdx) {
    return true;
  }
```

There is no other statement that can parse an `ID` token.